### PR TITLE
BZ2107507: Added note about checkup permissions

### PR DESCRIPTION
**Pull request by sjhala-ccs**
Version(s): 4.12+

Issue: https://bugzilla.redhat.com/show_bug.cgi?id=2107507

Link to docs preview: http://file.rdu.redhat.com/sjhala/BZ2107507/virt/logging_events_monitoring/virt-running-cluster-checkups.html#virt-about-cluster-checkup-framework_virt-running-cluster-checkups

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Review by phoracek — APPROVED** [review]
Looks good to me, but I would like to hear from @orelmisan who is much more familiar with security aspects of Kiagnose.

**Review by orelmisan — COMMENTED** [review]
Hi @sjhala-ccs, I've noticed that this PR has both 4.11 and 4.12 labels to it. This change is relevant to the 4.12 version.
The phrasing for the 4.11 version should be a bit different.

**Comment by sjhala-ccs** [comment]
> Hi @sjhala-ccs, I've noticed that this PR has both 4.11 and 4.12 labels to it. This change is relevant to the 4.12 version. The phrasing for the 4.11 version should be a bit different.

@orelmisan This bug was opened against the 4.11 docs rather than as a new story for 4.12. That is why I applied both labels to the PR. Can you provide the phrasing for 4.11 please? Thanks.

**Comment by orelmisan** [comment]
From the upstream [documentation](https://github.com/kiagnose/kiagnose/blob/release-0.1/README.md#checkup-installation) of the release matching 4.11:
> Warning: It is up to the cluster administrator to ALWAYS check the checkup's image and permissions BEFORE applying them and running the checkup. Kiagnose will AUTOMATICALLY bind these permissions to the checkup instance.

**Comment by sjhala-ccs** [comment]
> From the upstream [documentation](https://github.com/kiagnose/kiagnose/blob/release-0.1/README.md#checkup-installation) of the release matching 4.11:
> 
> > Warning: It is up to the cluster administrator to ALWAYS check the checkup's image and permissions BEFORE applying them and running the checkup. Kiagnose will AUTOMATICALLY bind these permissions to the checkup instance.

@orelmisan I am not sure how the meaning of the PR phrasing is different from the upstream text.

**Comment by orelmisan** [comment]
In 4.11, users input the names of existing ClusterRole objects they wish to use with the checkup.
Kiagnose will automatically bind these ClusterRole objects to the ServiceAccount it creates for the checkup instance.

In 4.12 the user provides a ServiceAccount object, pre-bound to whatever Role / ClusterRole required by the checkup.

**Comment by sjhala-ccs** [comment]
@orelmisan Does this PR for 4.11 look okay to you: https://github.com/openshift/openshift-docs/pull/51569. Thank you.

**Review by orelmisan — COMMENTED** [review]
Thank you for the change @sjhala-ccs.
The message of the warning should be IMO:
1. Always verify the checkup's image before running it.
2. Always review the checkup's permissions before creating the ClusterRole / Role object(s).
3. Always verify the name of the ServiceAccount you define in the user-supplied ConfigMap, because the framework automatically uses it for the checkup instance.

**Review by orelmisan — APPROVED** [review]
Thank you for the changes @sjhala-ccs.

**Comment by ousleyp** [comment]
LGTM!

File: modules/virt-about-cluster-checkup-framework.adoc [hunk 1]
```diff
@@ -10,4 +10,13 @@ A _checkup_ is an automated test workload that allows you to verify if a specifi
...
 
-Running a predefined checkup in the cluster involves setting up the namespace and service account for the framework, creating the `ClusterRole` and `ClusterRoleBinding` objects for the service account, enabling permissions for the checkup, and creating the input config map and the checkup job. You can run a checkup multiple times. 
+Running a predefined checkup in the cluster involves setting up the namespace and service account for the framework, creating the `ClusterRole` and `ClusterRoleBinding` objects for the service account, enabling permissions for the checkup, and creating the input config map and the checkup job. You can run a checkup multiple times.
+
+[IMPORTANT]
+====
+You must always:
+
+* Verify that the checkup image is from a trustworthy source before applying it.
+* Review the checkup permissions before creating the `ClusterRole` and `Role` objects.
```
Comment (yossisegev) [inline modules/virt-about-cluster-checkup-framework.adoc:20]: :s/Role/ClusterRoleBinding ?

**Review by yossisegev — CHANGES_REQUESTED** [review]

File: modules/virt-about-cluster-checkup-framework.adoc [hunk 2]
```diff
@@ -10,4 +10,13 @@ A _checkup_ is an automated test workload that allows you to verify if a specifi
...
+* Verify the name of the `ServiceAccount` in the config map. This is because the framework automatically binds these permissions to the checkup instance.
```
Comment (yossisegev) [inline modules/virt-about-cluster-checkup-framework.adoc:21]: `ServiceAccount` or `ClusterRole`?

**Comment by orelmisan** [comment]
@sjhala-ccs does the fact that PR #52386 reverts PR #50187 invalidates this PR as well?

**Comment by sjhala-ccs** [comment]
@orelmisan I accidentally merged https://github.com/openshift/openshift-docs/pull/50187 before peer review was complete. That is why I reverted the PR. Those changes will get merged after peer review, so I think this PR is still valid.

**Comment by orelmisan** [comment]
> @orelmisan I accidentally merged #50187 before peer review was complete. That is why I reverted the PR. Those changes will get merged after peer review, so I think this PR is still valid.

Thank you for clarifying.

File: modules/virt-about-cluster-checkup-framework.adoc (hunk 2 above)
Comment (sjhala-ccs) [reply modules/virt-about-cluster-checkup-framework.adoc:21]: @yossisegev If you are okay with the wording in this PR, can you please approve? Thank you.

**Pull request merged**